lib/Support/PathList.cpp

```diff
@@ -24,6 +27,32 @@ std::string getCurrentExecutableFullPath() {
   return FullPath.str().str();
 }
 
+std::string getCurrentRoot() {
+  using llvm::sys::path::parent_path;
+
+  std::ifstream Maps("/proc/self/maps");
```

Comment:
Let's avoid `ifstream`, let's use `MemoryBuffer::getFile` and then `StringRef.split("\n")`.

Reply:
Done